lib/spack/spack/util/executable.py

```diff
@@ -272,9 +276,10 @@ def __str__(self):
         return ' '.join(self.exe)
 
 
+@system_path_filter
 def which_string(*args, **kwargs):
     """Like ``which()``, but return a string instead of an ``Executable``."""
-    path = kwargs.get('path', os.environ.get('PATH', ''))
+    path = path_to_os_path(kwargs.get('path', os.environ.get('PATH', ''))).pop()
```

Comment:
Refactored to operate on a single path entry at a time. You are correct and the method would have handled the separated list, but I agree, its safer to avoid abusing it.